pkg/secrets/asm.go

```diff
@@ -22,15 +21,15 @@ func init() {
 	BackendRegister("asm", NewAWSSecretsManagerBackend)
 }
 
-func NewAWSSecretsManagerBackend() BackendIface {
+func NewAWSSecretsManagerBackend() Backend {
 	backend := &AWSSecretsManagerBackend{}
 	return backend
 }
 
-func (s *AWSSecretsManagerBackend) Init(params ...interface{}) error {
+func (s *AWSSecretsManagerBackend) Init(parameters map[string]string) error {
 	var err error
 
-	s.config, err = awsConfigFromParams(params...)
+	s.config, err = awsConfigFromParams(parameters)
```

Comment:
If we are using `map[string]string` as parameters then `awsConfigFromParams` and `paramsToMap` can be greatly simplified.

Some of the code in there just handles the conversion from `parameters ...interface{}` to `map[string]string`.

Now that's going to be unnecessary. Either remove it or create an issue and assign it to me.

Reply:
Fixed in https://github.com/ContainerSolutions/externalsecret-operator/pull/26/commits/2dd338830d194eb8f0d3436b8f0de281ca32792f